core/src/operations/decorators/mod.rs

```diff
@@ -0,0 +1,46 @@
+mod advice;
+use crate::utils::collections::Vec;
+pub use advice::AdviceInjector;
+use core::fmt;
+
+// DECORATORS
+// ================================================================================================
+
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub enum Decorator {
+    /// Injects zero or more values at the head of the advice tape as specified by the injector.
+    /// This operation affects only the advice tape, but has no effect on other VM components
+    /// (e.g., stack, memory), and does not advance VM clock.
+    Advice(AdviceInjector),
+}
+
+impl fmt::Display for Decorator {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::Advice(injector) => write!(f, "advice({})", injector),
+        }
+    }
+}
+
+pub type DecoratorList = Vec<(usize, Decorator)>;
+
+pub struct DecoratorIterator<'a> {
```

Comment:
Let's add come doc comments for both the struct as well as the methods (e.g., `next()`).

Reply:
Done